### PR TITLE
Properly toggle "permanent delete" check in confirmation dialog

### LINKED ISSUE #6849: Deleting files from WSL does not work if confirmation dialogue is turned off
### Description

Deleting files in WSL seems to not work if the confirmation dialogue option is turned off. It seems it still tries to do recycling instead of deleting even though that's not possible. 

Turning the dialogue back on resolves the issue since the checkbox for recycling is by default off in the dialogue and cannot be changed.

Additionally the folder view does not update in either of these cases - so the deleted files/folders are still visible after deleting and one has to go out of the directory and back in to see the changes.

### Steps To Reproduce

1. Go to settings and turn off "Show confirmation dialogue when deleting files or folders"
2. Go to WSL and try deleting a file or directory
3. An error message stating "Failed to move N items from \\wsl.localhost\..." appears in the task log in the upper right corner 

### Expected behavior

Deletion works with the dialogue options disabled and updates the folder view as well.

### Files Version

Version: 2.0.34.0 OS Version: 10.0.22000.282

### Windows Version

Windows 11 Version 21H2  OS build 22000.282 

### Relevant Assets

image

### LINKED ISSUE #7028: Failed tasks show check mark
### Description

image
I have seen this for multiple deletion failed from the recycle bin

### Steps To Reproduce

_No response_

### Expected behavior

It should show a X

### Files Version

2.0.36.0

### Windows Version

10.0.19043.1348

### Relevant Assets

[debug.log](https://github.com/files-community/Files/files/7581603/debug.log)
[debug_fulltrust.log](https://github.com/files-community/Files/files/7581604/debug_fulltrust.log)

### LINKED ISSUE #7334: Feature: Add setting to makes delete permanently delete items by default
### Description

Files are undeletable if I turn off "show a confirmation dialogue when deleting files"  but I still get a green checkmark for the operation.

### Steps To Reproduce

Go to recycle bin
click on properties.
turn on "dont move files to recycle bin"
go to files app
turn off the "show a confirmation dialogue" in preferences
try to delete a file

### Expected behavior

File gets deleted

### Files Version

Version: 2.0.34.0 OS Version: 10.0.22000.376

### Windows Version

Edition	Windows 11 Pro Version	21H2 Installed on	‎06/‎11/‎2021 OS build	22000.376 Experience	Windows Feature Experience Pack 1000.22000.376.0

### Relevant Assets/Logs

Example of the deletion process on files compared to files explorer
https://youtu.be/cCRKiRlO4HI

The ongoing tasks section with the green checkmark.
https://i.imgur.com/9WbyiEM.png

### LINKED ISSUE #7492: Files Attempts (and fails) to Move a File to the Recycle Bin When It Is Disabled by Windows
### Description

Files produces an error when attempting to delete a file when the Recycle bin is disabled. When the "Permanently Delete checkbox is turned on, files then properly delete, but having to check this every time is very inconvenient. This can be seen as a feature request, but I am filing as a "Bug" because it results in an error when the workaround isn't utilized.

**Thank you** for your continued support of Files

### Steps To Reproduce

Within Windows:
1. Right-Click on the "Recycle Bin" on the Desktop
2. Select "Properties"
3. Select "Don't move files to the Recycle Bin. Remove files immediately when deleted."

Within Files:
1. Find a file, and hit the "Delete" button.
2. In the delete panel, click "Delete" (without selecting "Permanently Delete)

### Expected behavior

The functionality is already built into files with the "Permanently Delete" checkbox. To make using Files more usable when the Recycle Bin is disabled, there are two ways that Files can better handle deleting a file:

1. The "Permanently Delete" checkbox should be automatically enabled (and user interaction should be disabled so it can't be un-checked as it simply results in an error).
2. A new option under the "Preferences" to automatically always enable the "Permanently Delete" checkbox when removing files.

### Files Version

Version: 2.0.48.0 OS Version: 10.0.22000.376

### Windows Version

Windows 11 22000.376

### Relevant Assets/Logs

Screenshot 2022-01-06 163911

### PR DESCRIPTION
**Pull request by gave92**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #7028
- Closes #7492
- Closes #7334
- Closes #6849

**Details of Changes**
Add details of changes here.
- Added a method to check whether deleting a file would move it to the recycle bin or not, the method e.g returns false in removable drives, network drives and if the recycle bin is disabled for a drive
- If recycle bin is not enabled, automatically check the "permanent delete" dialog option
- Call "UpdateBanner" when a banner is posted to fix #7028

Note: code in "FileUtils.cs" is currently not used and can be ignored, it's ground work to implement #1681.

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

File: src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs [hunk 1]
```diff
@@ -227,8 +314,19 @@ private async Task ParseFileOperationAsync(PipeStream connection, Dictionary<str
...
+                                    {
+                                        using var shi = new ShellItem(fileToDeletePath[i]);
+                                        op.QueueDeleteOperation(shi);
+                                    }))
+                                    {
+                                        shellOperationResult.Items.Add(new ShellOperationItemResult()
+                                        {
+                                            Succeeded = false,
+                                            Source = fileToDeletePath[i],
+                                            HRresult = (int)-1
```
Comment (d2dyno1) [inline src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs:327]: Unrelated, but Should be Hresult or HResult

File: src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs [hunk 2]
```diff
@@ -697,24 +833,26 @@ private void UpdateFileTagsDb(object sender, ShellFileOperations.ShellFileOpEven
...
+                var destPath = e.DestFolder != null ? (e.DestFolder.IsFileSystem ? e.DestFolder.FileSystemPath : e.DestFolder.ParsingName) : null;
                 var destination = operationType switch
                 {
-                    "delete" => e.DestItem?.FileSystemPath,
-                    "rename" => (!string.IsNullOrEmpty(e.Name) ? Path.Combine(Path.GetDirectoryName(e.SourceItem.FileSystemPath), e.Name) : null),
-                    "copy" => (e.DestFolder?.FileSystemPath != null && !string.IsNullOrEmpty(e.Name) ? Path.Combine(e.DestFolder.FileSystemPath, e.Name) : null),
-                    _ => (e.DestFolder?.FileSystemPath != null && !string.IsNullOrEmpty(e.Name) ? Path.Combine(e.DestFolder.FileSystemPath, e.Name) : null)
+                    "delete" => e.DestItem != null ? (e.DestItem.IsFileSystem ? e.DestItem.FileSystemPath : e.DestItem.ParsingName) : null,
+                    "rename" => !string.IsNullOrEmpty(e.Name) ? Path.Combine(Path.GetDirectoryName(sourcePath), e.Name) : null,
+                    "copy" => destPath != null && !string.IsNullOrEmpty(e.Name) ? Path.Combine(destPath, e.Name) : null,
```
Comment (d2dyno1) [inline src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs:842]: !string.IsNullOrEmpty(destPath) ?

File: src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs [hunk 3]
```diff
@@ -155,17 +155,28 @@ private async Task ParseFileOperationAsync(PipeStream connection, Dictionary<str
...
+                                }
 
                                 var createTcs = new TaskCompletionSource<bool>();
                                 op.PostNewItem += (s, e) =>
                                 {
                                     shellOperationResult.Items.Add(new ShellOperationItemResult()
                                     {
                                         Succeeded = e.Result.Succeeded,
-                                        Destination = e.DestItem?.FileSystemPath,
+                                        Destination = e.DestItem != null ? (e.DestItem.IsFileSystem ? e.DestItem.FileSystemPath : e.DestItem.ParsingName) : null,
```
Comment (d2dyno1) [inline src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs:179]: Can some kind of helper method be written to avoid duplicated checks?

**Review by d2dyno1 — COMMENTED** [review]
The code looks good! I've added some *very* picky things to my review

File: src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs (hunk 1 above)
Comment (gave92) [reply src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs]: Thanks lol never noticed

File: src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs (hunk 3 above)
Comment (gave92) [reply src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs]: Done! Looks a lot better now xD

File: src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs (hunk 2 above)
Comment (gave92) [reply src/Files.Launcher/MessageHandlers/FileOperationsHandler.cs:839]: Not necessary, the path will either be valid or null.

**Review by d2dyno1 — APPROVED** [review]
Looks good!

**Pull request merged**